### DISCUSSION TITLE
`separator` in `csv.decode()`?

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#399] [Q&A] [answered]
@pgundlach.

I play with `https://github.com/speedata/examples/blob/master/technical/csvreader/`.

I wanted to read from CSV files with `;` as separator. Once replacing comma with semicolon and saving he CSV file, I updated `csv.decode("us-500.csv", {separator = ";"})` in `csv2xml.lua`.

All I get is the following error:

````
record on line 2: wrong number of fields
````

What am I missing to make the separator work?

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
Syntax look correct, I think you need to post the fist lines of the CSV.

**Reply by pr-apes** [reply]
Here you have them:

````
"First name";"Last name";"Company name";"Address";"City";"County";"State";"ZIP";"Phone1";"Phone2";"E-mail";"Web"
"James";"Butt";"Benton, John B Jr";"6649 N Blue Gum St";"New Orleans";"Orleans";"LA",70116,"504-621-8927";"504-845-1427";"jbutt@gmail.com";"http://www.bentonjohnbjr.com"
"Josephine";"Darakjy";"Chanay, Jeffrey A Esq";"4 B Blue Ridge Blvd";"Brighton";"Livingston";"MI",48116,"810-292-9388";"810-374-9840";"josephine_darakjy@darakjy.org";"http://www.chanayjeffreyaesq.com"
"Art";"Venere";"Chemel, James L Cpa";"8 W Cerritos Ave #54";"Bridgeport";"Gloucester";"NJ";"08014";"856-636-8749";"856-264-4130";"art@venere.org";"http://www.chemeljameslcpa.com"
````

**Reply by pgundlach** [reply]
This does not look like a correct CSV file to me. I see other separators than ;

**Reply by pr-apes** [reply]
Sorry, but I’m not sure I’m getting your point.

If fields (or cells) are enclosed in quotes, commas can be used inside them and they should not considered separators.

Also, if a given separator is defined, I think other characters should not be considered separators.

Or what am I missing here?

**Comment by pgundlach** [comment]
`"LA",70116,` the comma is not within the quotes.

**Reply by pr-apes** [reply]
Sorry, it was my fault.
